File: MTMA.API/MTMA.API/MTMA.API/Controllers/IdentityController.cs
namespace MTMA.API.Controllers
{
    using Microsoft.AspNetCore.Authorization;
    using Microsoft.AspNetCore.Mvc;
    using MTMA.Services.Identity;
    using MTMA.Services.ServiceModels;

    [ApiController]
    [Route("api/[controller]")]
    public class IdentityController : ControllerBase
    {
        private readonly IIdentityService _identityService;

        public IdentityController(IIdentityService identityService)
            => this._identityService = identityService;

        [HttpPost]
        [Route(nameof(Register))]
        public async Task<IActionResult> Register(RegisterUserServiceModel serviceModel)
        {
            var identityResult = await this._identityService.Register(serviceModel);

            return identityResult.Succeeded
                ? this.Ok()
                : this.BadRequest(identityResult);
        }

        [HttpPost]
        [Route(nameof(Login))]
        public async Task<IActionResult> Login([FromBody] LoginUserServiceModel serviceModel)
        {
            var (identityResult, token) = await this._identityService.Login(serviceModel);

            return identityResult.Succeeded
                ? this.Ok(new { token = token })
                : this.BadRequest(identityResult);
        }

        [HttpPost]
        [Authorize]
        [Route(nameof(ChangePassword))]
        public async Task<IActionResult> ChangePassword([FromBody] ChangePasswordServiceModel serviceModel)
        {
            var identityResult = await this._identityService.ChangePassword(serviceModel);

            return identityResult.Succeeded
                ? this.Ok()
                : this.BadRequest(identityResult);
        }
    }
}

File: MTMA.UI/mock_tests/lib/ui/controllers/identity_controller.dart
import 'package:mock_tests/core/service_models/identity/login_response_service_model.dart';
import 'package:mock_tests/core/service_models/identity/login_service_model.dart';
import 'package:mock_tests/core/service_models/identity/register_error_details_service_model.dart';
import 'package:mock_tests/core/service_models/identity/register_user_service_model.dart';
import 'package:mock_tests/core/services/identity/identity_service_abstract.dart';

class IdentityController {
  static final IdentityController _instance = IdentityController._();

  static late final IdentityServiceAbstract _identityService;

  IdentityController._();
  factory IdentityController(IdentityServiceAbstract identityService) {
    _identityService = identityService;
    return _instance;
  }

  Future<RegisterErrorDetailsServiceModel> register(RegisterServiceModel serviceModel) async {
    return await _identityService.register(serviceModel);
  }

  Future<LoginResponseServiceModel> login(LoginServiceModel serviceModel) async {
    return await _identityService.login(serviceModel);
  }
}
